src/core/translation.ml

```diff
@@ -177,6 +177,8 @@ let rec xpost_pattern ~driver exn = function
       ppat_construct (lident exn)
         (Some (ppat_var (noloc (str "%a" Tterm.Ident.pp x.vs_name))))
   | Tterm.Papp (ls, []) when Tterm.(ls_equal ls (fs_tuple 0)) -> pvar exn
+  | Tterm.Papp (ls, l) when Tterm.is_fs_tuple ls ->
+      ppat_construct (lident exn) (Some (ppat_tuple (List.map pattern l)))
```

Comment:
Here we are building the arguments of the exception, not sure why we would want its name ?
This will generate something like:
```ocaml
let y =
     try f x
     with
     | E _ as __e__003_ ->
         ((match __e__003_ with
           | E (E s, E i) ->
```

Reply:
Ah right, sorry for the noise!